test/exceptions/src/assert_fail.d

```diff
@@ -60,6 +60,29 @@ void testFloatingPoint()()
     test(creal(1 + 2i), creal(-1 + 2i), "1 + 2i != -1 + 2i");
 }
 
+void testPointers()
+{
+    static struct S
+    {
+        string toString() const { return "S(...)"; }
+    }
+
+    static if ((void*).sizeof == 4)
+        enum ptr = "12345678";
+    else
+        enum ptr = "123456789012345";
+
+    int* p = cast(int*) mixin("0x" ~ ptr);
+    string msg = _d_assert_fail!"=="(cast(S*) p, p);
+
+    // Some sprintf implementations prepend 0 or 0x
```

Comment:
Ugh. Would it make sense to just print the equivalent `size_t` as hex so it's consistent? We don't have to be slaves to the vagaries of C implementations.

i.e. I would prefer:

```D
sprintf("0x%x", p); // use appropriate lx or llx if needed
```

Reply:
Note this is especially important for implementations that don't print `0x`, as if they do not contain any letters (as your example doesn't) it's not clear that it's hex.

Reply:
Done.